### ISSUE TITLE
When overridding logging handlers for a specific category we get NullPointerException in native mode only

### ISSUE THREAD

**Issue by debae** [quarkusio/quarkus#6954]
**Describe the bug**
Whenever we configure specific handlers that we link to a category we get a NullPointerException.
```
2020-01-30 12:10:53 Mounting /private/var/folders/xk/xtn2b68s2c33qryl26pvzfv40000gn/T/tmpt7c35ugv as /var/task:ro,delegated inside runtime container
Exception in thread "main" java.lang.RuntimeException: Failed to start quarkus
at io.quarkus.runner.ApplicationImpl.doStart(ApplicationImpl.zig:138)
at io.quarkus.runtime.Application.start(Application.java:87)
at io.quarkus.runtime.Application.run(Application.java:210)
at io.quarkus.runner.GeneratedMain.main(GeneratedMain.zig:41)
Caused by: java.lang.NullPointerException
at org.jboss.logmanager.AtomicArray.add(AtomicArray.java:114)
at org.jboss.logmanager.LoggerNode.addHandler(LoggerNode.java:295)
at org.jboss.logmanager.Logger.addHandler(Logger.java:217)
at io.quarkus.runtime.logging.LoggingSetupRecorder.addNamedHandlersToCategory(LoggingSetupRecorder.java:169)
at io.quarkus.runtime.logging.LoggingSetupRecorder.initializeLogging(LoggingSetupRecorder.java:116)
at io.quarkus.deployment.steps.LoggingResourceProcessor$setupLoggingRuntimeInit180.deploy_0(LoggingResourceProcessor$setupLoggingRuntimeInit180.zig:103)
at io.quarkus.deployment.steps.LoggingResourceProcessor$setupLoggingRuntimeInit180.deploy(LoggingResourceProcessor$setupLoggingRuntimeInit180.zig:122)
at io.quarkus.runner.ApplicationImpl.doStart(ApplicationImpl.zig:60)
... 3 more
```

**Expected behavior**
The behaviour should be the same as when running in JVM and attach the extra handlers to the Logger that belongs to a category.

**To Reproduce**
1. Configure named handlers and link to category
2. Build native image
3. During initialisation the exception occurs

**Configuration**
```properties
quarkus.log.handler.console."STRUCTURED_LOGGING".format=%m%n

quarkus.log.category."com.xyz.DocumentEventLogger".level=DEBUG
quarkus.log.category."com.xyz.DocumentEventLogger".handlers=STRUCTURED_LOGGING
```

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: Darwin APRXJGH6D5D5F9 18.7.0 Darwin Kernel Version 18.7.0: Thu Jan 23 06:52:12 PST 2020; root:xnu-4903.278.25~1/RELEASE_X86_64 x86_64
 - Output of `java -version`: 1.8.0_211
 - GraalVM version (if different from Java): quay.io/quarkus/ubi-quarkus-native-image:19.2.1
 - Quarkus version or git rev: 1.2.0.Final

**Comment by gsmet** [comment]
@dmlloyd looks like an interesting puzzle for you!

**Comment by debae** [comment]
Already has a chat with him, his response was : 
```
David Lloyd: OK I think I can infer that the problem is here: io.quarkus.runtime.graal.Target_org_jboss_logmanager_LoggerNode

David Lloyd: we're basically nulling that field in the native image heap image

David Lloyd: perhaps we should change that to be FromAlias and then assign a constant empty Handler array to it
```

I'm trying this now and see if that solves the issue.

**Comment by debae** [comment]
I've made some changes : 
```
Index: core/runtime/src/main/java/io/quarkus/runtime/graal/LoggingSubstitutions.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- core/runtime/src/main/java/io/quarkus/runtime/graal/LoggingSubstitutions.java	(revision dc3b0f7767d468c6b382ec6ec69469bdeb05aff4)
+++ core/runtime/src/main/java/io/quarkus/runtime/graal/LoggingSubstitutions.java	(date 1580808446000)
@@ -20,8 +20,8 @@
 final class Target_org_jboss_logmanager_LoggerNode {
 
     @Alias
-    @RecomputeFieldValue(kind = RecomputeFieldValue.Kind.Reset)
-    volatile Handler[] handlers;
+    @RecomputeFieldValue(kind = RecomputeFieldValue.Kind.FromAlias)
+    volatile Handler[] handlers = LoggingSubstitutions.EMPTY_HANDLERS_ARRAY;
 }
 
 @TargetClass(className = "org.slf4j.LoggerFactory")
@@ -55,4 +55,7 @@
 }
 
 final class LoggingSubstitutions {
+
+    static final Handler[] EMPTY_HANDLERS_ARRAY = new Handler[0];
+
 }

``
`

But this gave a NullPointerException while building the native image which I was not able to figure out why

**Comment by phact** [comment]
Getting a similar (almost identical) dump on startup for native-image running `1.2.0-Final`

```
target/lunar-lander-1.0.0-SNAPSHOT-runner 
Exception in thread "main" java.lang.RuntimeException: Failed to start quarkus
	at io.quarkus.runner.ApplicationImpl.doStart(ApplicationImpl.zig:194)
	at io.quarkus.runtime.Application.start(Application.java:87)
	at io.quarkus.runtime.Application.run(Application.java:210)
	at io.quarkus.runner.GeneratedMain.main(GeneratedMain.zig:41)
Caused by: java.lang.NullPointerException
	at io.quarkus.runtime.logging.LoggingSetupRecorder.createNamedHandlers(LoggingSetupRecorder.java:138)
	at io.quarkus.runtime.logging.LoggingSetupRecorder.initializeLogging(LoggingSetupRecorder.java:105)
	at io.quarkus.deployment.steps.LoggingResourceProcessor$setupLoggingRuntimeInit10.deploy_0(LoggingResourceProcessor$setupLoggingRuntimeInit10.zig:103)
	at io.quarkus.deployment.steps.LoggingResourceProcessor$setupLoggingRuntimeInit10.deploy(LoggingResourceProcessor$setupLoggingRuntimeInit10.zig:122)
	at io.quarkus.runner.ApplicationImpl.doStart(ApplicationImpl.zig:84)
```

**Comment by phact** [comment]
In case this is helpful, it works when I roll back to `1.1.0.CR1`

**Comment by debae** [comment]
@phact Can you please add the configuration of you application.yaml , because the error is different according to me.
The reason why you don't get this exception in 1.1.0.CR1 is because the named handlers are only introduced as of version 1.2.0.

**Comment by schulzp** [comment]
Hi, I run into the same issue, even when running the `app.jar` inside a dockerized JVM (`Dockerfile.jvm`), see my [MWE](https://github.com/schulzp/quarkus-logging-npe).

**Comment by dmlloyd** [comment]
@schulzp could you please share your Quarkus version & the stack trace you are seeing?

**Comment by schulzp** [comment]
@dmlloyd, sure, my [minimal working example](https://github.com/schulzp/quarkus-logging-npe) uses quarkus 1.2.0 and right after running the -runner.jar [inside a container] with [Open]JDK 11 the error occurs:

```text
Exception in thread "main" java.lang.RuntimeException: Failed to start quarkus
	at io.quarkus.runner.ApplicationImpl.doStart(ApplicationImpl.zig:210)
	at io.quarkus.runtime.Application.start(Application.java:87)
	at io.quarkus.runtime.Application.run(Application.java:210)
	at io.quarkus.runner.GeneratedMain.main(GeneratedMain.zig:41)
Caused by: java.lang.NullPointerException
	at io.quarkus.runtime.logging.LoggingSetupRecorder.createNamedHandlers(LoggingSetupRecorder.java:138)
	at io.quarkus.runtime.logging.LoggingSetupRecorder.initializeLogging(LoggingSetupRecorder.java:105)
	at io.quarkus.deployment.steps.LoggingResourceProcessor$setupLoggingRuntimeInit2.deploy_0(LoggingResourceProcessor$setupLoggingRuntimeInit2.zig:103)
	at io.quarkus.deployment.steps.LoggingResourceProcessor$setupLoggingRuntimeInit2.deploy(LoggingResourceProcessor$setupLoggingRuntimeInit2.zig:122)
	at io.quarkus.runner.ApplicationImpl.doStart(ApplicationImpl.zig:68)
	... 3 more
```

**Comment by schulzp** [comment]
@dmlloyd, I updated my [minimal (not-)working example](https://github.com/schulzp/quarkus-logging-npe) to quarkus 1.2.1.Final and the NPE still occurs with the same stack trace.

**Comment by schulzp** [comment]
Resolved: NPE was due to different versions for `quarkus.platform.version` and `quarkus-plugin.version and no longer occurs.

**Comment by thankthemaker** [comment]
Any progress on this? I tested it with the rest-json-quickstart from the Quarkus quickstarts (with quarkus version 1.2.0 and 1.4.1) and the NPE is still there.

**Comment by adryen31** [comment]
Hello all,
for me I have the same problem, but even if @schulzp 's idea, it not working ... however I have the same version for quarkus.platform.version and quarkus-plugin.version I dispair because it was for a POC in my company in order to switch from Spring Boot to Quarkus, but even if my leader team's vision may not be able to convince the technical director
![plugin](https://user-images.githubusercontent.com/1407536/110198508-65c92b00-7e53-11eb-98ab-4a1aeaf7b31f.png)
![log](https://user-images.githubusercontent.com/1407536/110198509-6661c180-7e53-11eb-9015-a3025134dff5.png)
![dependency](https://user-images.githubusercontent.com/1407536/110198510-66fa5800-7e53-11eb-801e-ede6c646ae99.png)
![stacktrace](https://user-images.githubusercontent.com/1407536/110198732-e9cfe280-7e54-11eb-900a-3e2b882ef81a.png)

But for me the NPE was generated in the "org.jboss.logmanager.AtomicArray.add(AtomicArray.java:114)"
Maybe I forgotten anything in configuration ?
Anyone have another idea by any chance plz ?
Do you think I can pull the source code for debugging ?

Thank you all

**Comment by jaikiran** [comment]
>But for me the NPE was generated in the "org.jboss.logmanager.AtomicArray.add(AtomicArray.java:114)"

There's a PR that's currently being tested which intends to fix this https://github.com/quarkusio/quarkus/pull/15578

**Comment by debae** [comment]
This issue is solved in 1.13.0.Final by @dmlloyd
